ocaml/xapi/console.ml

```diff
@@ -126,7 +127,309 @@ let address_of_console __context console : address option =
     ) ;
   address_option
 
-let real_proxy' vnc_port s =
+let log_in_file ?(file = "/var/log/idle_timeout.txt") msg =
+  let oc = open_out_gen [Open_creat; Open_text; Open_append] 0o644 file in
+  let time_str =
+    let tm = Unix.localtime (Unix.time ()) in
+    Printf.sprintf "%04d-%02d-%02d %02d:%02d:%02d" (tm.tm_year + 1900)
+      (tm.tm_mon + 1) tm.tm_mday tm.tm_hour tm.tm_min tm.tm_sec
+  in
+  output_string oc (Printf.sprintf "[%s] %s\n" time_str msg) ;
+  close_out oc
+
+(* https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst *)
+module RFB_msg_type_parser = struct
+  type state = {
+      mutable rfb_phase: [`Handshake | `Security | `ClientInit | `Messages]
```

Comment:
Can use `variants` instead of `polymorphic variants`?
And I believe there is one phase `No_parsing` for the case that parsing failed. It's impossible to recover. The session should be exempted from the idle timeout checking.